effekt/shared/src/main/scala/effekt/core/Transformer.scala

```diff
@@ -41,14 +41,19 @@ object Transformer extends Phase[Typechecked, CoreTransformed] {
     ModuleDecl(path, mod.imports.map { _.path }, declarations, externals, optimized, exports)
   }
 
+  def addToScope(definition: Definition, body: Stmt): Stmt = body match {
+    case Scope(definitions, body) => Scope(definition :: definitions, body)
+    case other => Scope(List(definition), other)
+  }
```

Comment:
We use this helper function (for now) to not introduce too many `Scope`s in the translation.